src/jit/SimdInl.h

```diff
@@ -311,3 +311,37 @@ static void emitGlobalSet128(sljit_compiler* compiler, Instruction* instr)
     sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0, SLJIT_MEM1(SLJIT_R0), context->globalsStart + globalSet->index() * sizeof(void*));
     sljit_emit_simd_mov(compiler, SLJIT_SIMD_STORE | SLJIT_SIMD_REG_128 | SLJIT_SIMD_ELEM_128, src.arg, SLJIT_MEM1(SLJIT_R0), JITFieldAccessor::globalValueOffset());
 }
+
+void emitSelect128(sljit_compiler* compiler, Instruction* instr, sljit_s32 jump_type)
+{
+    Operand* operands = instr->operands();
+    assert(instr->opcode() == ByteCode::SelectOpcode && instr->paramCount() == 3);
+
+    JITArg args[2];
+
+
+    JITArg target(operands + 3);
+    sljit_s32 dstReg = GET_TARGET_REG(target.arg, SLJIT_FR0);
+
+    sljit_s32 arg1_reg = sljit_has_cpu_feature(SLJIT_SIMD_REGS_ARE_PAIRS) ? SLJIT_FR2 : SLJIT_FR1;
+
+    simdOperandToArg(compiler, operands, args[0], SLJIT_SIMD_ELEM_128, SLJIT_FR0);
+    simdOperandToArg(compiler, operands + 1, args[1], SLJIT_SIMD_ELEM_128, arg1_reg);
+
+    const sljit_s32 mov_type = SLJIT_SIMD_ELEM_128 | SLJIT_SIMD_REG_128;
+
+    sljit_jump* jump = NULL;
+
+    if (jump_type == -1) {
+        JITArg cond(operands + 2);
+        jump = sljit_emit_cmp(compiler, SLJIT_NOT_EQUAL, cond.arg, cond.argw, SLJIT_IMM, 0);
+    } else {
+        jump = sljit_emit_jump(compiler, jump_type);
+    }
+    sljit_emit_simd_mov(compiler, mov_type, dstReg, args[1].arg, args[1].argw);
```

Comment:
I think two movs are needed for select

Reply:
Yeah, this is why I asked if `simdOperandToArg` can be relied on to always move its argument to a register.  If it can, then there is no need to `mov` in one branch.  Otherwise yes, there should be two `mov`s.  (Which is how I originally implemented it.)